### PR TITLE
[532] Add DISCLAIMER file to META-INF folder in jars

### PR DESCRIPTION
**Pull request by rangareddy**
## *Important Read*

This pull request aims to address issues [#532](https://github.com/apache/incubator-xtable/issues/532)

## What is the purpose of the pull request

Adding the DISCLAIMER file to META-INF folders in all jars.

## Brief change log

* Added the two new properties to <properties> tag in pom.xml

 ```xml
<apache-jar-resource-bundle.version>1.7</apache-jar-resource-bundle.version>
<apache-incubator-disclaimer-resource-bundle.version>1.7</apache-incubator-disclaimer-resource-bundle.version>
```
* Added the `<resourceBundles>` to the `maven-remote-resources-plugin` plugin in pom.xml file.

```xml
<resourceBundles>
    <resourceBundle>org.apache.apache.resources:apache-jar-resource-bundle:${apache-jar-resource-bundle.version}</resourceBundle>
    <resourceBundle>org.apache.apache.resources:apache-incubator-disclaimer-resource-bundle:${apache-incubator-disclaimer-resource-bundle.version}</resourceBundle>
</resourceBundles>
```

## Verify this pull request

Run `$ mvn clean install -DskipTests -U` locally and extract the jars and verify `META-INF` folder.

**META-INF folder before this PR:**

```sh
% ls -ltr META-INF 
total 48
drwxr-xr-x  3 rangareddy  staff     96 Jul  4 18:52 maven
-rw-r--r--  1 rangareddy  staff    176 Jul  4 18:52 NOTICE
-rw-r--r--  1 rangareddy  staff    341 Jul  4 18:52 MANIFEST.MF
-rw-r--r--  1 rangareddy  staff  11358 Jul  4 18:52 LICENSE
-rw-r--r--  1 rangareddy  staff   1942 Jul  4 18:52 DEPENDENCIES
```

**META-INF folder after this PR:**

```sh
% ls -ltr META-INF
total 56
drwxr-xr-x  3 rangareddy  staff     96 Jul  4 18:52 maven
-rw-r--r--  1 rangareddy  staff    179 Jul  4 18:52 NOTICE
-rw-r--r--  1 rangareddy  staff    341 Jul  4 18:52 MANIFEST.MF
-rw-r--r--  1 rangareddy  staff  11358 Jul  4 18:52 LICENSE
-rw-r--r--  1 rangareddy  staff    559 Jul  4 18:52 DISCLAIMER
-rw-r--r--  1 rangareddy  staff   1942 Jul  4 18:52 DEPENDENCIES
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -570,6 +570,10 @@
                             <properties>
                                 <projectName>Apache XTable (incubating)</projectName>
                             </properties>
+                            <resourceBundles>
+                                <resourceBundle>org.apache:apache-jar-resource-bundle:1.4</resourceBundle>
+                                <resourceBundle>org.apache.apache.resources:apache-incubator-disclaimer-resource-bundle:1.2-SNAPSHOT</resourceBundle>
```
[the remark this answers is not in the document]
Comment (rangareddy) [reply pom.xml:575]: The below artifacts are old and it is last updated on Mar 08, 2007. 
https://mvnrepository.com/artifact/org.apache/apache-incubator-disclaimer-resource-bundle

You can refer following article for latest artifacts for various apache resource bundles.

- https://maven.apache.org/apache-resource-bundles/index.html
- https://maven.apache.org/apache-resource-bundles/incubator-disclaimer/ 
- https://mvnrepository.com/artifact/org.apache.apache.resources

File: DISCLAIMER [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Apache XTable™ (Incubating) is an effort undergoing incubation at the Apache Software
+Foundation (ASF), sponsored by the Apache Incubator PMC.
+
+Incubation is required of all newly accepted projects until a further review
```
[the remark this answers is not in the document]
Comment (rangareddy) [reply DISCLAIMER]: DISCLAIMER file automatically generated by using following template: 

https://github.com/apache/maven-apache-resources/blob/apache-resource-bundles-1.7/incubator-disclaimer/src/main/resources/META-INF/DISCLAIMER.vm

**Comment by rangareddy** [comment]
Hi @vinishjail97 Review the latest commit changes.

**Comment by vinishjail97** [comment]
cat META-INF/DISCLAIMER 
```
Apache XTable (incubating) is an effort undergoing incubation at the Apache Software 
Foundation (ASF), sponsored by the Apache Incubator PMC. 

Incubation is required of all newly accepted projects until a further review 
indicates that the infrastructure, communications, and decision making process 
have stabilized in a manner consistent with other successful ASF projects. 

While incubation status is not necessarily a reflection of the completeness 
or stability of the code, it does indicate that the project has yet to be 
fully endorsed by the ASF.
```

@pjfanning Can you confirm if this looks okay ? We are still keeping DISCLAIMER-WIP in the project though.

**Review by vinishjail97 — APPROVED** [review]

**Comment by vinishjail97** [comment]
@rangareddy We can merge after @pjfanning gives an approval as well.

**Comment by pjfanning** [comment]
I think you should add DISCLAIMER-WIP to the jar instead of DISCLAIMER while the project uses the DISCLAIMER-WIP and change to DISCLAIMER when the project switches to DISCLAIMER.

**Comment by vinishjail97** [comment]
> I think you should add DISCLAIMER-WIP to the jar instead of DISCLAIMER while the project uses the DISCLAIMER-WIP and change to DISCLAIMER when the project switches to DISCLAIMER. The exact same file that you have in the project - not some other file.

I had the same doubt but the resource bundle provided by Apache for incubating projects doesn't have that ability.
https://maven.apache.org/apache-resource-bundles/incubator-disclaimer/ 
https://github.com/apache/maven-apache-resources/blob/master/incubator-disclaimer/src/main/resources/META-INF/DISCLAIMER.vm

**Review by pjfanning — APPROVED** [review]
Approving. Having the basic DISCLAIMER is better than having none.

**Pull request merged**